### PR TITLE
macOS: Bundle dSYM packages in FlutterMacOS.xcframework

### PR DESCRIPTION
**Pull request by cbracken**
As of Xcode 16, App Store validation requires dSYMs for frameworks in app archives. Bundling dSYMs also significantly simplifies stack trace symbolification, so we should be doing this regardless.

This adds both framework and simulator framework dSYMs to the FlutterMacOS.xcframework bundle.

Issue: https://github.com/flutter/flutter/issues/153879

## Pre-launch Checklist

- [X] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [X] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [X] I read and followed the [Flutter Style Guide] and the [C++, Objective-C, Java style guides].
- [X] I listed at least one issue that this PR fixes in the description above.
- [X] I added new tests to check the change I am making or feature I am adding, or the PR is [test-exempt]. See [testing the engine] for instructions on writing and running engine tests.
- [X] I updated/added relevant documentation (doc comments with `///`).
- [X] I signed the [CLA].
- [X] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[C++, Objective-C, Java style guides]: https://github.com/flutter/engine/blob/main/CONTRIBUTING.md#style
[testing the engine]: https://github.com/flutter/flutter/wiki/Testing-the-engine
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by cbracken** [comment]
Once presubmits have completed, I'll send out a test roll to the framework.

File: sky/tools/sky_utils.py
Comment (cbracken) [inline sky/tools/sky_utils.py:87]: Note: just like `-f` checks in shell script, `os.path.isfile` will return `True` when passed the path to a symlink that resolves to a file.

File: sky/tools/sky_utils.py
Comment (cbracken) [inline sky/tools/sky_utils.py:131]: Note the name change from `FlutterMacOS.dSYM` to `FlutterMacOS.framework.dSYM` here. In the `zip_xcframework_archive` function, we do the same [disgusting hack](https://github.com/flutter/engine/pull/54458/files#diff-09e320619e9e7e0a91fabf0255b5bb0edb9d4a3dbe7937dfdb2646288ec9f41cR222-R239) for backward-compatibility we did for iOS before the tool side change landed, and we removed the need for the archive altogether (#54492).

File: sky/tools/create_macos_framework.py
Comment (jmagman) [inline sky/tools/create_macos_framework.py:110]:
```suggestion
  # Historically, the framework dSYM was named FlutterMacOS.dSYM, so in order to
```

File: sky/tools/create_macos_framework.py
Comment (jmagman) [inline sky/tools/create_macos_framework.py:138]: Can you add a comment to this and the iOS one says if this list is ever updated, also update <url to cocoon test>? That way the release team doesn't have a surprise.

**Comment by cbracken** [comment]
Framework test roll out in https://github.com/flutter/flutter/pull/153898.

File: sky/tools/create_macos_framework.py
Comment (jmagman) [reply sky/tools/create_macos_framework.py:138]: Sorry, I was writing from my phone and misremembered, the test isn't in cocoon, it's in flutter/flutter:
https://github.com/flutter/flutter/blob/62382c7b83a16b3f48dc06c19a47f6b8667005a5/dev/bots/suite_runners/run_verify_binaries_codesigned_tests.dart#L88-L108

I think that list needs to stay up-to-date with this one.  So when this PR rolls, that test should be updated to avoid release issues like https://github.com/flutter/flutter/pull/153787

Comment (cbracken) [reply sky/tools/create_macos_framework.py:138]: Awesome! Also on my phone but will look in a bit! Thanks for the pointer.

**Comment by cbracken** [comment]
Test roll succeeded: https://github.com/flutter/flutter/pull/153898. I'll add the comments and push an update here.

File: sky/tools/create_macos_framework.py
Comment (cbracken) [reply sky/tools/create_macos_framework.py:149]: Done.

File: sky/tools/create_macos_framework.py
Comment (cbracken) [reply sky/tools/create_macos_framework.py]: Done!

**Review by zanderso — APPROVED** [review]

**Pull request merged**
